app/controllers/public_controller.rb

```diff
@@ -20,7 +20,7 @@ def profile
     @posts = @profile.posts
     @my_comments = Comment.where(account_id: @profile.id)
     @my_posts = Post.where(account_id: @profile.id, is_drafted: false).page(params[:page]).per 5
-    @hot_myposts = Post.where(account_id: @profile.id).order(view_count: :desc).page(params[:page]).per 5
+    @hot_myposts = Post.where(account_id: @profile.id).order(upvotes: :desc).page(params[:page]).per 5
     @top_myposts = Post.where(account_id: @profile.id).order(view_count: :desc).page(params[:page]).per 5
     @new_myposts = Post.where(account_id: @profile.id).order(created_at: :desc).page(params[:page]).per 5
```

Comment:
Can you refactor this, please? use separate partials for each tab and also do it via ajax @tusharravindran

Reply:
In Progress .Will Resolve it soon

Reply:
Can we do it as a seperate PR @iMagesh due to same type of code in [Index]
![image](https://user-images.githubusercontent.com/57782379/212549124-60aab2a5-82e3-4ef5-83ed-a9346546db19.png)

Reply:
I will do it as a separate Branch and send PR